src/caffe/data_transformer.cpp

```diff
@@ -1,4 +1,6 @@
+#ifdef USE_OPENCV
 #include <opencv2/core/core.hpp>
+#endif
```

Comment:
This file should be called `image_data_transformer`. It can't be applied to text, audio and other types of data.